### PR TITLE
Load resources from entities' catalog item loader.

### PR DESCRIPTION
**Pull request by neykov**

### CODE REVIEW

File: usage/camp/src/test/java/io/brooklyn/camp/brooklyn/catalog/CatalogYamlEntityTest.java [hunk 1]
```diff
@@ -362,6 +364,20 @@ public void testCreateSpecFromCatalogItem() {
...
+        String id = "resource.test";
+        addCatalogOSGiEntity(id, SIMPLE_ENTITY_TYPE);
+        String yaml =
+                "services: \n" +
+                "  - serviceType: "+ver(id);
+        Entity app = createAndStartApplication(yaml);
+        Entity simpleEntity = Iterables.getOnlyElement(app.getChildren());
+        InputStream icon = new ResourceUtils(simpleEntity).getResourceFromUrl("classpath:/brooklyn/osgi/tests/icon.gif");
+        assertTrue(icon != null);
+        icon.close();
```
Comment (sjcorbett) [inline usage/camp/src/test/java/io/brooklyn/camp/brooklyn/catalog/CatalogYamlEntityTest.java]: Close in a finally block?

Comment (neykov) [inline usage/camp/src/test/java/io/brooklyn/camp/brooklyn/catalog/CatalogYamlEntityTest.java]: Not really needed as there's nothing to throw between opening and closing it. `getResourceFromUrl` will throw if the resource is not found, but I've added the additional `assertTrue(icon != null);` just to illustrate intent.

Comment (sjcorbett) [inline usage/camp/src/test/java/io/brooklyn/camp/brooklyn/catalog/CatalogYamlEntityTest.java]: Ah, I was thrown by the illustration of intent.

Comment (neykov) [inline usage/camp/src/test/java/io/brooklyn/camp/brooklyn/catalog/CatalogYamlEntityTest.java]: :) Well the assert will throw only on null, so the close won't be needed in this case.

**Comment by sjcorbett** [comment]
My comment was irrelevant so looks good. Merging.

**Pull request merged**